R/swagger.R

```diff
@@ -124,54 +200,125 @@ extractResponses <- function(resps){
 #' Extract the swagger-friendly parameter definitions from the endpoint
 #' paramters.
 #' @noRd
-extractSwaggerParams <- function(endpointParams, pathParams){
+extractSwaggerParams <- function(endpointParams, pathParams, funcParams = NULL){
+
+  params <- list(
+    parameters = list(),
+    requestBody = list()
+  )
+  inBody <- filterSwaggerTypes("requestBody", "location")
+  inRaw <- filterSwaggerTypes("binary", "format")
+  for (p in unique(c(names(endpointParams), pathParams$name, names(funcParams)))) {
+
+    # Dealing with priorities endpointParams > pathParams > funcParams
+    # For each p, find out which source to trust for :
+    #   `type`, `isArray`, `required`
+    # - `description` comes from endpointParams
+    # - `isArray` defines both `style` and `explode`
+    # - `default` and `example` comes from funcParams
+    # - `location` change to "path" when p is in pathParams and
+    #   unused when `type` is "object" or "file"
+    # - When type is `object`, create a requestBody with content
+    #   default to "application/json"
+    # - When type is `file`, change requestBody content to
+    #   multipart/form-data
 
-  params <- list()
-  for (p in names(endpointParams)) {
-    location <- "query"
     if (p %in% pathParams$name) {
       location <- "path"
+      required <- TRUE
+      style <- "simple"
+      explode <- FALSE
+      type <- priorizeProperty(defaultSwaggerType,
+                               pathParams[pathParams$name == p,]$type,
+                               endpointParams[[p]]$type,
+                               funcParams[[p]]$type)
```

Comment:
Could store this info at the start of the loop and then refer in multiple places

Reply:
This is done one time when plumber is first ran. Any reason beside performance? This has to be done for each parameter individually at the moment. PriorizeProperty was not made to handle all of them at once.

Reply:
Just performance, not a blocking issue